pkg/devfile/adapters/kubernetes/component/adapter.go

```
@@ -240,109 +241,220 @@ func substitueYamlVariables(baseYaml []byte, yamlSubstitutions map[string]string
+								newService["spec"].(map[string]interface{})["clusterIP"] = currentServiceSpec["clusterIP"]
+								deploymentManifest.SetUnstructuredContent(newService)
+							}
+						}
+						instanceFound = true
+					}
+				}
+			}
+
+			s := log.Spinnerf("Deploying the manifest for %s", gvk.Kind)
+			result := &unstructured.Unstructured{}
+			actionType := "create"
+			if !instanceFound {
+				result, err = a.Client.DynamicClient.Resource(gvr).Namespace(namespace).Create(deploymentManifest, metav1.CreateOptions{})
+			} else {
+				actionType = "update" // Update deployment
+				result, err = a.Client.DynamicClient.Resource(gvr).Namespace(namespace).Update(deploymentManifest, metav1.UpdateOptions{})
+			}
+			if err != nil {
+				s.End(false)
+				return errors.Wrapf(err, "Failed to %s manifest %s", actionType, gvk.Kind)
+			} else {
+				s.End(true)
+				log.Successf("%sd manifest for %s (%s)", strings.Title(actionType), applicationName, gvk.Kind)
 			}
+
+			// Write the returned manifest to the local manifest file
+			if writtenToManifest {
+				_, err = manifestFile.WriteString("---\n")
+				if err != nil {
+					return errors.Wrap(err, "Unable to write to local manifest file")
+				}
+			}
+			err = yamlDecoder.Encode(result, manifestFile)
+			if err != nil {
+				return errors.Wrap(err, "Unable to write to local manifest file")
+			}
+			writtenToManifest = true
 		}
 	}
 
-	result := &unstructured.Unstructured{}
-	if !instanceFound {
-		// Create Deployment
-		log.Infof("Creating %s...", gvk.Kind)
-		result, err = myclient.Resource(gvr).Namespace(namespace).Create(deploymentManifest, metav1.CreateOptions{})
-		//	result, err := a.Client.DynamicClient.Resource(gvr).Namespace(namespace).Create(deploymentManifest, metav1.CreateOptions{})
-	} else {
-		// Update Deployment
-		log.Infof("Updating %s...", gvk.Kind)
-		result, err = myclient.Resource(gvr).Namespace(namespace).Update(deploymentManifest, metav1.UpdateOptions{})
-	}
+	_ = manifestFile.Close()
```

we would want to return this error. I would rather write a defer where we create the files as such:
```[golang]
defer func() {
      merr := manifestFile.Close()
      if err == nil {
            err = merr
      }
}()

---

Agree